packages/atomic/src/components/facets/facet-search/facet-search.tsx

```diff
@@ -50,30 +45,45 @@ export class FacetSearch {
       onSelectValue: (element) => {
         const value = (element as HTMLLIElement).value;
         if (value === FacetSearch.ShowMoreResultsValue) {
-          return this.facetSearch.showMoreResults();
+          return this.facetSearchController.showMoreResults();
         }
-        this.onSelectValue(this.values[value]);
+        this.onSelectValue(value);
       },
       onBlur: () => (this.component.showFacetSearchResults = false),
       activeClass: 'active',
-      activePartName: 'active-suggestion',
+      activePartName: 'active-search-result',
     });
   }
 
+  public static highlightSearchResult(resultValue: string, query: string) {
+    const sanitizedResult = sanitize(resultValue);
+    if (query.trim() === '') {
+      return sanitizedResult;
+    }
+
+    const search = regexEncode(query);
+    const regex = new RegExp(`(${search})`, 'ig');
+    return sanitize(resultValue).replace(regex, '<b>$1</b>');
+  }
+
+  public static get searchResultClasses() {
+    return 'search-result cursor-pointer px-2 py-1 text-sm flex flex-col justify-center';
+  }
+
   public updateCombobox() {
     this.combobox.updateAccessibilityAttributes();
   }
 
-  private get facetSearch() {
+  private get facetSearchController() {
     return this.component.facet.facetSearch;
   }
 
   private get facetSearchState() {
     return this.component.facet.state.facetSearch;
   }
 
-  private get values(): FacetSearchResult[] {
-    return this.component.facet.state.facetSearch.values;
+  private get facetSearchResults() {
+    return this.component.facet.state.facetSearch.values as FacetSearchResult[];
```

Comment:
Why do we need to do a cast here ?

Reply:
because of https://github.com/coveo/ui-kit/pull/701#discussion_r612584329